File: jinwoo/11주차/추억 점수.py
# 시간 : 5분50초
# Lv : 1

def solution(name, yearning, photo):
    dic = {}
    for i in range(len(name)):
        dic[name[i]] =yearning[i]
    
    answer=[]
    for ph in photo:
        sum=0
        for each_name in ph:
            if each_name in dic:
                sum+=dic[each_name]
        answer.append(sum)
        
    return answer

# 파이썬 개사기. 문법 몰라도 이게되네 

File: jHyuk/week11/추억 점수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;

    map<string, int> m;

    for (int i = 0; i < name.size(); i++) {
        m[name[i]] = yearning[i];
    }

    for (int i = 0; i < photo.size(); i++) {
        int sum = 0;
        for (int j = 0; j < photo[i].size(); j++) {
            if (m.find(photo[i][j]) != m.end()) {
                sum += m[photo[i][j]];
            }
        }
        answer.push_back(sum);
    }

    return answer;
}